library/alloc/src/string.rs

```diff
@@ -1969,8 +1970,11 @@ impl String {
     ///
     /// It does not reallocate or shrink the `String`,
     /// so the leaked allocation may include unused capacity that is not part
-    /// of the returned slice. If you don't want that, call [`into_boxed_str`],
-    /// and then [`Box::leak`].
+    /// of the returned slice. The advantage of this is that it avoids the potential
+    /// reallocation that [`shrink_to_fit`] may perform.
```

Comment:
Something feels very odd in terms wording here to me. When I read that there's no reallocations, then I wouldn't expect a sentence saying that it's somehow an advantage over some other seemingly unrelated method to leaking.

Reply:
Yeah, you're not wrong. Gave it another try :+1: